internal/controllers/user_devices_controller.go

```diff
@@ -231,28 +231,26 @@ func (udc *UserDevicesController) dbDevicesToDisplay(ctx context.Context, device
 		var nft *VehicleNFTData
 		var credential *VINCredentialData
 		pu := []PrivilegeUser{}
-
-		if vnft := d.R.VehicleNFT; vnft != nil {
-			nftStatus := vnft.R.MintRequest
+		if !d.TokenID.IsZero() {
```

Comment:
Unfortunately I think we have to be more careful here. There can still be interesting information in this section even if minting isn't finished (and so a token id is present).